apps/cms/src/customize/graphql/mutation/inviteTest.ts

```diff
@@ -0,0 +1,210 @@
+              $and: [{ candidate: { email }, test: { id: testId } }],
+            },
+          });
+
+        if (!isIndTestExist) {
+          const currentCandidate = await strapi.db
+            .query('api::candidate.candidate')
+            .findOne({ where: { email } });
+
+          // todo currently createMany doesn't work correctly, find a way to use
+          await strapi.db.query('api::individual-test.individual-test').create({
+            data: {
+              test: testId,
+              candidate: currentCandidate.id,
+              publishedAt: new Date().toISOString(),
+            },
+          });
+        }
+      })
+    );
+  } catch (error) {
+    throw new ApplicationError(error.message);
+  }
+};
+
+const getIndTests = async (emails: string[], testId: string) => {
+  try {
+    const listOfIndTest = await Promise.all(
+      emails.map(async (email: string) => {
+        const indTest = await strapi.db
+          .query('api::individual-test.individual-test')
+          .findOne({
+            where: {
+              $and: [{ candidate: { email }, test: { id: testId } }],
+            },
+          });
+
+        const token = getService('jwt').issue({
+          individualTestId: indTest.id,
+          SECRET_KEY: process.env.JWT_SECRET,
+        });
+
+        return {
+          email,
+          // todo replace with real link-test
+          linkTestInvite: `http://localhost:4200/test/${testId}?token=${token}`,
+        };
+      })
+    );
+
+    return listOfIndTest;
+  } catch (error) {
+    throw new ApplicationError(error.message);
+  }
+};
+
+const sendEmails = async (listOfIndTests: IndividualTest[]) => {
+  await Promise.all(
+    listOfIndTests.map(async (item) => {
+      await strapi.plugins['email'].services.email.sendTemplatedEmail(
+        {
+          to: item.email,
+          from: process.env.SENDGRID_EMAIL_FROM,
+          replyTo: process.env.SENDGRID_EMAIL_FROM,
+        },
+        emailTemplate,
+        {
+          candidate: _.startCase(_.toLower(item.email.split(/\.|@/)[0])),
+          linkTestInvite: item.linkTestInvite,
+        }
+      );
+    })
+  );
+};
+
+/**
+ * HANDLE INVITE CANDIDATES
+ * @param _
+ * @param args
+ */
+export const inviteTest = async (_, args) => {
+  const { testId, emails } = args;
+
+  /**
+   * validate testId and emails
+   */
+
+  if (await validateTestId(testId)) {
+    return {
+      status: 'failed',
+      message: 'TestId is not valid.',
+    };
+  }
+
+  if (validateEmails(emails)) {
+    return {
+      status: 'failed',
+      message: 'Emails input is not valid.',
+    };
+  }
+
+  /**
+   * Check if candidate belong to email is exists then create non-exist candidate by the email
+   */
+  const listCandidateToCreate: Candidate[] = await getNonExistingCandidates(
+    emails
+  );
+  await createManyCandidate(listCandidateToCreate);
+
+  /**
+   * Save data to Individual test
+   */
+  await createIndividualTests(emails, testId);
```

Comment:
To improve query 
const candidates = await strapi.db
          .query('api::candidate.candidate')
          .findMany({ where: { emails } }) -> At this time we all emails is created candidate
`createIndividualTests(candidateIds, testId)`
` await getIndTests(candidateIds, testId)`